src/rtHttpRequest.cpp

```diff
@@ -237,6 +238,7 @@ void rtHttpRequest::onDownloadComplete(rtFileDownloadRequest* downloadRequest)
   } else {
     req->mEmit.send("error", downloadRequest->errorString());
   }
+  req->Release();
```

Comment:
Hiding a rtHttpRequest `Release()` inside of `onDownloadComplete` is dangerous as it decrements the reference without the programmer's knowledge.  All APIs that register for `onDownloadComplete` now need to call `AddRef()` before registering for this callback.  Instead, please add a new callback function to rtHttpRequest called `onDownloadCompleteAndRelease`.  This callback will simply call rtHttpRequest's `onDownloadComplete` and then it will call `Release()` on the rtHttpRequest that is part of the passed in rtFileDownloadRequest's callback data.  The method should look like this:

`
void rtHttpRequest::onDownloadCompleteAndRelease(rtFileDownloadRequest* downloadRequest)
{
  onDownloadComplete(downloadRequest);
  rtHttpRequest* req = (rtHttpRequest*)downloadRequest->callbackData();
  req->Release();
}
`